packages/eslint-plugin/lib/rules/no-unused-vars.js

```diff
@@ -92,6 +92,12 @@ module.exports = Object.assign({}, baseRule, {
       },
       '*[declare=true] Identifier'(node) {
         context.markVariableAsUsed(node.name);
+        const scope = context.getScope();
+        const { variableScope } = scope;
+        if (variableScope !== scope) {
+          const superVar = variableScope.set.get(node.name);
+          if (superVar) superVar.eslintUsed = true;
+        }
```

Comment:
we should execture `context.markVariableAsUsed` in `else`

Reply:
I wanted to keep both, since `markVariableAsUsed` modifies the class’s scope, while my additions modify the module scope.